docs/backend-system/building-backends/08-migrating.md

```diff
@@ -7,3 +7,238 @@ description: How to migrate existing backends to the new backend system
+The second argument allows you to "tweak" the types of those services to
+something more fitting to your env. For example, you'll see that the logger
+service API type was changed from the raw Winston logger of old, to a different,
+custom API, so we use a helper function to transform that particular one.
+
+To make additions as mentioned above to the environment, you will start to get
+into the weeds of how the backend system wiring works. You'll need to have a
+service reference, and a service factory that performs the actual creation of
+your service. For now, let's add them directly in the same file, and then you
+can move them out into a separate file at a later time.
+
+In this example, we'll assume that your added environment item is named "example".
+
+```diff
+-import { coreServices } from '@backstage/backend-plugin-api';
++import { coreServices, createServiceFactory } from '@backstage/backend-plugin-api';
++import { ExampleApi, ExampleImpl } from '<somewhere>';
+
++const exampleServiceRef = createServiceRef<ExampleApi>({
++  id: 'example',
++  scope: 'plugin', // can be 'root' or 'plugin'
++  defaultFactory: async service => createServiceFactory({
++    service,
++    // optional; just an example of how dependencies work
++    deps: {
++      logger: coreServices.logger,
++    },
++    // here instances of the declared dependencies are injected
++    async factory({ logger }) {
++      // return your ExampleImpl
++    },
++  }),
++});
+
+ const legacyPlugin = makeLegacyPlugin(
+   {
+     // ... the above core services still go here
++    example: exampleServiceRef
+   },
+   {
+     logger: log => loggerToWinstonLogger(log),
+   },
+ );
+```
+
+After this, your backend will know how to instantiate your thing on demand.
+
+# Cleaning Up the Plugins Folder
+
+For plugins that are private and your own, you can follow a [dedicated migration
+guide](../building-plugins-and-modules/08-migrating.md) as you see fit, at a
+later time.
+
+For third party backend plugins, in particular the larger core plugins that are
+maintained by the Backstage maintainers, you may find that they have already
+been migrated to the new backend system. Let's try to clean up the `plugins`
+folder a bit.
+
+Each plugin file is under your control, and it was added as part of following
+their installation instructions. You were then free to start tweaking those
+files and adding customizations. The simplest case is when you did NOT perform
+any customizations at all. For those plugins, you can in general just delete the
+corresponding file in the `plugins` folder, and instead importing the backend
+plugin from the package directly.
+
+```diff
+ import { createBackend } from '@backstage/backend-defaults';
+ import { legacyPlugin } from '@backstage/backend-common';
++import { catalogPlugin } from '@backstage/plugin-catalog-backend';
+
+ const backend = createBackend();
+ backend.add(legacyPlugin('todo', import('./plugins/todo')));
++backend.add(catalogPlugin());
```

Comment:
This would break scaffolder templates since the scaffolder module installation is handled by the plugin setup file in the old system. I think we'll need to create some form of doc pattern for migrating individual plugins. Maybe we should even have a later section here that briefly describes how to migrate all core plugins? Their extension points, etc.

Reply:
Sure, I initially wanted to use the todo plugin here as an example but it was actually not converted yet. I'm rewriting it all to cover all converted plugins now :) Brb